lib/ssl/src/tls_connection.erl

```diff
@@ -580,24 +580,29 @@ init({call, From}, {start, Timeout},
             handshake_env = #handshake_env{renegotiation = {Renegotiation, _}} = HsEnv,
             connection_env = CEnv,
 	    ssl_options = #{log_level := LogLevel,
-                            %% Use highest version in initial ClientHello.
-                            %% Versions is a descending list of supported versions.
-                            versions := [HelloVersion|_] = Versions,
-                            session_tickets := SessionTickets} = SslOpts,
-	    session = NewSession,
-	    connection_states = ConnectionStates0
-	   } = State0) ->
+                        %% Use highest version in initial ClientHello.
+                        %% Versions is a descending list of supported versions.
+                        versions := [HelloVersion|_] = Versions,
+                        session_tickets := SessionTickets,
+                        ocsp_stapling := OcspStaplingOpt,
+                        ocsp_responder_certs := OcspResponderCerts,
+                        ocsp_nonce := OcspNonceOpt} = SslOpts,
+	                    session = NewSession,
+	                    connection_states = ConnectionStates0
+	    } = State0) ->
     KeyShare = maybe_generate_client_shares(SslOpts),
     Session = ssl_session:client_select_session({Host, Port, SslOpts}, Cache, CacheCb, NewSession),
     %% Update UseTicket in case of automatic session resumption
     {UseTicket, State1} = tls_handshake_1_3:maybe_automatic_session_resumption(State0),
     TicketData = tls_handshake_1_3:get_ticket_data(self(), SessionTickets, UseTicket),
+    OcspNonce = public_key:ocsp_nonce(OcspNonceOpt, OcspStaplingOpt),
```

Comment:
I had a comment about to wrap this function, I can't find it anymore, but it was about not to leak the abstraction of the public_key API. `public_key:ocsp_nonce`should not be aware of any variables from applications using the public_key API. So `public_key:ocsp_nonce/0` should just return a DER encoded nonce, no parameters needed.

Reply:
done